codegen/method.go

```diff
@@ -1153,7 +1162,7 @@ func (ms *MethodSpec) setWriteQueryParamStatements(
 			if field.Required {
 				statements.appendf("if r%s == nil {", longFieldName)
 				// TODO: generate correct number of nils...
-				statements.append("\treturn nil, nil, errors.New(")
+				statements.appendf("\treturn %snil, errors.New(", nilResultStr)
```

Comment:
I see why you are doing this but I think an if...else would be more readable here.

Reply:
Since this is not in perf path, it is reasonable to do so and... I should remove the old TODO.